docs/conf.py

```diff
@@ -46,6 +47,8 @@
     "sphinxcontrib.autoprogram",
 ]
 
+autosectionlabel_prefix_document = True
```

Comment:
This adds prefixes to each label, using the document name, reducing risks of conflicts.

https://www.sphinx-doc.org/en/master/usage/extensions/autosectionlabel.html